code/ndarray.c

```diff
@@ -78,7 +78,7 @@ void ndarray_print_row(const mp_print_t *print, mp_obj_array_t *data, size_t n0,
         }
         mp_printf(print, ", ..., ");
         mp_obj_print_helper(print, mp_binary_get_val_array(data->typecode, data->items, n0+n-3), PRINT_REPR);
-        for(size_t i=1; i<3; i++) {
+        for(i=1; i<3; i++) {
```

Comment:
Must the declaration of `i` really go? The old version made `i` local to that particular loop, which I find easier to work with.

Reply:
The compiler is probably warning that the `size_t i=1;` on line 81 is shadowing `size_t i;` line 66. The line 66 declaration could be removed and the `for` loops could each have their own declaration.

Reply:
This is actually my bad, thanks for pointing this out. You are right, line 66 should just be removed. Actually, in the long run it won't be needed at all: https://github.com/v923z/micropython-ulab/blob/2867ec9f13024167fcb22fb06a2e46b1e7fea46d/code/ndarray.c#L76

Reply:
cool, good to know, I'll switch it around in my fork

Reply:
Before you are carrier away, that is the ndim branch:) For now, we should just yank line 66, and add `size_t i=` in the for loops.